perceval/backends/jenkins.py

```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#     Alvaro del Castillo <acs@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import CacheError
+from ..utils import (DEFAULT_DATETIME,
+                     datetime_to_utc,
+                     str_to_datetime,
+                     urljoin)
+
+logger = logging.getLogger(__name__)
+
+class Jenkins(Backend):
+    """Jenkins backend for Perceval.
+
+    This class retrieves the builds from a
+    Jenkins site. To initialize this class the
+    site must be provided.
+
+    :param site: Jenkins site
+    :param job_url: Jenkins job from which to get the builds
+    :param cache: cache object to store raw data
+    :param origin: identifier of the repository; when `None` or an
+        empty string are given, it will be set to `site` value
+    """
+    version = '0.1.0'
+
+    def __init__(self, site, job_url=None, cache=None, origin=None):
```

Comment:
I have another comment here regarding `job_url`. I understand what you are trying to do but I think it breaks the common schema that we have on the rest of backends. We don't implement the possibility of fetching data from an item and if we allow it, the functionality should be implemented to retrieve data from a list of items, not for just one item.